node/core/pvf/src/prepare/worker.rs

```diff
@@ -273,44 +273,42 @@ where
 
 async fn send_request(
 	stream: &mut UnixStream,
-	code: Arc<Vec<u8>>,
+	pvf_with_params: PvfWithExecutorParams,
 	tmp_file: &Path,
-	executor_params: &ExecutorParams,
 	preparation_timeout: Duration,
 ) -> io::Result<()> {
-	framed_send(stream, &code).await?;
+	framed_send(stream, &pvf_with_params.encode()).await?;
 	framed_send(stream, path_to_bytes(tmp_file)).await?;
-	framed_send(stream, &executor_params.encode()).await?;
 	framed_send(stream, &preparation_timeout.encode()).await?;
 	Ok(())
 }
 
 async fn recv_request(
 	stream: &mut UnixStream,
-) -> io::Result<(Vec<u8>, PathBuf, ExecutorParams, Duration)> {
-	let code = framed_recv(stream).await?;
+) -> io::Result<(PvfWithExecutorParams, PathBuf, Duration)> {
+	let pvf_with_params = framed_recv(stream).await?;
+	let pvf_with_params =
+		PvfWithExecutorParams::decode(&mut &pvf_with_params[..]).map_err(|e| {
```

Comment:
Have you tested decoding into an `Arc`? I saw your question about encoding `Arc<T>` (`T` gets encoded), but does the other way work as expected too?

Reply:
Yes, I did, it works.